ext/odbc/php_odbc.c

```diff
@@ -2518,13 +2545,12 @@ PHP_FUNCTION(odbc_field_num)
 	odbc_result *result;
 	zval *pv_res;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "rs", &pv_res, &fname, &fname_len) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "Os", &pv_res, odbc_result_ce, &fname, &fname_len) == FAILURE) {
 		RETURN_THROWS();
 	}
 
-	if ((result = (odbc_result *)zend_fetch_resource(Z_RES_P(pv_res), "ODBC result", le_result)) == NULL) {
-		RETURN_THROWS();
-	}
+	result = Z_ODBC_RESULT_P(pv_res);
+	CHECK_ODBC_RESULT(result);
```

Comment:
Drive by remark for bellow, the ``strcasecmp()`` call looks fishy as it doesn't deal with binary strings which might be passed as the ``fname`` value from userland